pkg/gardenlet/controller/shoot/shoot_care_control.go

```diff
@@ -277,13 +279,22 @@ func (c *defaultCareControl) Care(shootObj *gardencorev1beta1.Shoot, key string)
 		return nil
 	}
 
+	// Only read Garden secrets once because we don't rely on up-to-date secrets for health checks.
+	if c.gardenSecrets == nil {
+		secrets, err := garden.ReadGardenSecrets(ctx, gardenClient.Cache(), c.k8sGardenCoreInformers.Seeds().Lister(), seedpkg.ComputeGardenNamespace(*shoot.Spec.SeedName))
+		if err != nil {
+			return fmt.Errorf("error reading Garden secrets: %w", err)
+		}
+		c.gardenSecrets = secrets
+	}
+
```

Comment:
Out of curiosity, why are the secrets required for the health check? Can we potentially simply initialize the `Operation` with `nil`?

Reply:
I think the secrets themselves are not required but we need it to construct the operation. Otherwise, we'll have the following error:
```
time="2021-03-12T14:56:01+01:00" level=error msg="could not initialize a new operation: unable to figure out which secret should be used for dns"
```

I'm also not too happy with it but I think using the whole operation object to perform health checks comes at a price which we can see here. It was btw. one of the reasons I wanted to invest more time to refactor our health checks but unfortunately I haven't had enough capacity yet.

Reply:
OK, I see... 😢 We should definitely optimize this to make it less heavy-weight and to save ourselves coding like this.
Thanks for explaining!